src/Raven.Client/Documents/DocumentStore.cs

```diff
@@ -344,24 +345,74 @@ public override IDisposable AggressivelyCacheFor(TimeSpan cacheDuration, string
         /// without touching the server.
         /// </remarks>
         public override IDisposable AggressivelyCacheFor(TimeSpan cacheDuration, AggressiveCacheMode mode, string database = null)
+        {
+            var release = SetAggressiveCache(cacheDuration, mode, ref database);
+
+            return AsyncHelpers.RunSync(() => FinalizeAggressiveCacheAsync(release, mode, database));
+        }
+
+        /// <summary>
+        /// Setup the context for aggressive caching.
+        /// </summary>
+        /// <remarks>
+        /// Aggressive caching means that we will not check the server to see whether the response
+        /// we provide is current or not, but will serve the information directly from the local cache
+        /// without touching the server.
+        /// </remarks>
+        public override Task<IDisposable> AggressivelyCacheForAsync(TimeSpan cacheDuration, string database = null)
+        {
+            return AggressivelyCacheForAsync(cacheDuration, Conventions.AggressiveCache.Mode, database);
+        }
+
+        /// <summary>
+        /// Setup the context for aggressive caching.
+        /// </summary>
+        /// <remarks>
+        /// Aggressive caching means that we will not check the server to see whether the response
+        /// we provide is current or not, but will serve the information directly from the local cache
+        /// without touching the server.
+        /// </remarks>
+        public override Task<IDisposable> AggressivelyCacheForAsync(TimeSpan cacheDuration, AggressiveCacheMode mode, string database = null)
```

Comment:
Why doesn't this use `ValueTask`? It seems the majority case is that you are already connected and then you could entirely avoid the task allocation